auth0/src/main/java/com/auth0/android/authentication/storage/BaseCredentialsManager.kt

```diff
@@ -90,4 +85,30 @@ public abstract class BaseCredentialsManager internal constructor(
     protected fun hasExpired(expiresAt: Long): Boolean {
         return expiresAt <= currentTimeInMillis
     }
+}
+
+public abstract class DefaultCredentialsManager internal constructor(
+    authenticationClient: AuthenticationAPIClient,
+    storage: Storage,
+    jwtDecoder: JWTDecoder
+) : BaseCredentialsManager(
+    authenticationClient, storage, jwtDecoder
+) {
+    public abstract fun getCredentials(callback: Callback<Credentials, CredentialsManagerException>)
+    public abstract fun getCredentials(
+        scope: String?,
+        minTtl: Int,
+        callback: Callback<Credentials, CredentialsManagerException>
+    )
+}
+
+public abstract class SecuredCredentialsManager internal constructor(
+    authenticationClient: AuthenticationAPIClient,
+    storage: Storage,
+    jwtDecoder: JWTDecoder
+) : BaseCredentialsManager(
+    authenticationClient, storage, jwtDecoder
+) {
+    public abstract fun getCredentials(fragmentActivity: FragmentActivity, authenticationOptions: LocalAuthenticationOptions, callback: Callback<Credentials, CredentialsManagerException>)
```

Comment:
If FragmentActivity is required in both method, can't we have this as a class level dependency, that way you can have same abstract method for both cred managers.

We just have to ensure there is no context leak by having fragment activity at a class level

Reply:
Agreed, updated the code accordingly, please take a look again.